textattack/attack_recipes/kuleshov_2017.py

```diff
@@ -0,0 +1,61 @@
+"""
+    Kuleshov, V. et al. 
+    
+    Generating Natural Language Adversarial Examples. 
+    
+    
+    https://openreview.net/pdf?id=r1QZ3zbAZ.
+"""
+
+from textattack.constraints.overlap import WordsPerturbed
+from textattack.constraints.grammaticality.language_models import GPT2
+from textattack.constraints.semantics import ThoughtVector
+from textattack.goal_functions import UntargetedClassification
+from textattack.search_methods import GreedyWordSwap
+from textattack.transformations.black_box import WordSwapEmbedding
+
+def Kuleshov2017(model):
+    #
+    # "Specifically, in all experiments, we used a target of τ = 0.7,
+    # a neighborhood size of N = 15, and parameters λ_1 = 0.2 and δ = 0.5; we set
+    # the syntactic bound to λ_2 = 2 nats for sentiment analysis"
+    
+    #
+    # Word swap with top-15 counter-fitted embedding neighbors.
+    #
+    transformation = WordSwapEmbedding(max_candidates=15)
+    #
+    # Maximum of 50% of words perturbed (δ in the paper).
+    #
+    constraints = []
+    constraints.append(
+            WordsPerturbed(max_percent=0.5)
+    )
+    #
+    # Maximum thought vector Euclidean distance of λ_1 = 0.2. (eq. 4)
+    #
+    constraints.append(
+        ThoughtVector(embedding_type='paragramcf', max_mse_dist=0.2)
+    )
+    #
+    #
+    # Maximum language model log-probability difference of λ_2 = 0.2. (eq. 5)
+    #
+    constraints.append(
+        GPT2(max_log_prob_diff=2.0)
+    )
+    #
+    # Goal is untargeted classification: reduce original probability score 
+    # to below τ = 0.7 (Algorithm 1).
```

Comment:
Don't you have to pass in .7 then as an argument in line 51?

Reply:
yes :)